eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/RegisterSearchEnginePass.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * This file is part of the eZ Publish Kernel package
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+use LogicException;
+
+/**
+ * This compiler pass will register eZ Publish search engines
+ */
+class RegisterSearchEnginePass implements CompilerPassInterface
+{
+    /**
+     * Container service id of the SearchEngineFactory
+     *
+     * @see \eZ\Bundle\EzPublishCoreBundle\ApiLoader\SearchEngineFactory
+     *
+     * @var string
+     */
+    protected $factoryId = 'ezpublish.api.search_engine.factory';
```

Comment:
Actually it is probably going to change (ref discussion on search decoupling PR) :)
I changed this to constructor injection in https://github.com/ezsystems/ezpublish-kernel/commit/7044dc3f754205de372942b724fa3003cc144db5.